ubuntu24.04/Dockerfile

```
@@ -64,8 +64,11 @@ RUN echo "TARGETARCH=$TARGETARCH"
 ADD install.sh /tmp
 
 # Fetch GPG keys for CUDA repo
-RUN apt-key del 7fa2af80 && OS_ARCH=${TARGETARCH/amd64/x86_64} && OS_ARCH=${OS_ARCH/arm64/sbsa} && \
-    apt-key adv --fetch-keys "https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH}/3bf863cc.pub"
+RUN apt-key del 7fa2af80 3bf863cc && \
+    apt-get update && apt-get install -y --no-install-recommends curl gpg && \
+    OS_ARCH=${TARGETARCH/amd64/x86_64} && OS_ARCH=${OS_ARCH/arm64/sbsa} && \
+    curl -fsSL https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH}/3bf863cc.pub | gpg --dearmor -o /etc/apt/keyrings/cuda-3bf863cc.pub && \
+    echo "deb [signed-by=/etc/apt/keyrings/cuda-3bf863cc.pub] https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH} /" > /etc/apt/sources.list.d/cuda.list
```

Just a nit, but why not this? Just looks a tad better IMO
```suggestion
    echo "deb [signed-by=/etc/apt/keyrings/cuda.pub] https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH} /" > /etc/apt/sources.list.d/cuda.list
```

---

Updated to just `cuda.pub`